experimental/devices/ht16k33/ht16k33.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package ht16k33
+
+import (
+	"errors"
+
+	"periph.io/x/periph/conn/i2c"
+)
+
+// I2CAddr i2c default address.
+const I2CAddr uint16 = 0x70
+
+const (
+	cmdRAM        = 0x00
+	cmdKeys       = 0x40
+	displaySetup  = 0x80
+	displayOff    = 0x00
+	displayOn     = 0x01
+	systemSetup   = 0x20
+	oscillatorOff = 0x00
+	oscillatorOn  = 0x01
+	cmdBrightness = 0xE0
+)
+
+// BlinkFrequency display frequency must be a value allowed by the HT16K33.
+type BlinkFrequency byte
+
+// Note that frequency must be a value allowed by the HT16K33, specifically one of:
+const (
+	BlinkOff    = 0x00
+	Blink2Hz    = 0x02
+	Blink1Hz    = 0x04
+	BlinkHalfHz = 0x06
+)
+
+// Dev is a handler to ht16k33 controller
+type Dev struct {
+	dev *i2c.Dev
+}
+
+// NewI2C returns a Dev object that communicates over I2C.
+//
+// To use on the default address, ht16k33.I2CAddr must be passed as argument.
+func NewI2C(bus i2c.Bus, address uint16) (*Dev, error) {
+	dev := &Dev{
+		dev: &i2c.Dev{Bus: bus, Addr: address},
+	}
+	err := dev.init()
+	if err != nil {
+		return nil, err
+	}
+
+	return dev, nil
+}
+
+func (d *Dev) init() error {
+	// Turn on the oscillator.
+	if _, err := d.dev.Write([]byte{systemSetup | oscillatorOn}); err != nil {
+		return err
+	}
+
+	// Turn on display
+	if _, err := d.dev.Write([]byte{displaySetup | displayOn}); err != nil {
+		return err
+	}
+
+	// Set no blinking.
+	if err := d.SetBlink(BlinkOff); err != nil {
```

Comment:
I'd recommend to inline the write here with the commands before, instead of the method call. Same for SetBrightness.